src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -1149,6 +1153,29 @@ private String execDockerCmd(String command, int timeout) {
         return result;
     }
 
+    // Suggest a performance improvement if docker build takes too long.
+    private static final long DOCKER_BUILD_SOFT_TIMEOUT = 30000;
+    private void checkDockerIgnore(long startTime) {
+        if (System.currentTimeMillis() - startTime < DOCKER_BUILD_SOFT_TIMEOUT) {
+            return;
+        }
+        File dockerfileToUse = dockerfile != null ? dockerfile : defaultDockerfile;
+        if (dockerfileToUse.exists()) {
+            File dockerContext = dockerfileToUse.getParentFile();
+            debug("checkDockerIgnore, dockerContext="+dockerContext.getAbsolutePath());
+            File dockerIgnore = new File(dockerContext, ".dockerignore");
+            if (!dockerIgnore.exists()) { // provide some advice
```

Comment:
Should we suggest adding the runtime directories to `.dockerignore` even if they already have a `.dockerignore` file? I suppose this could get annoying if they already have the runtime dirs in `.dockerignore`, but if they don't, this would still be good advice.

Reply:
I think if they are using a `.dockerignore` they already know to search around for large directories to add to the file.